### PR TITLE
Cordova 4 updates

### PR DESCRIPTION
**Pull request by mobiRic**
Good day, Arne,

I decided to use your Parse plugin for my latest Ionic / Cordova project, and there were a few minor updates that needed to be done for it to work for me. Some of the updates stem from the fact that I am using Cordova 4.
1. I updated the Cordova plugin requirement to v4. This is because of some changes in CordovaWebView that I needed to make, which break backwards compatibility on older versions of Cordova
2. the Cordova Google Play plugin has been changed, and the old one no longer works with Cordova 4
3. the ParsePlugin initialisation call was not returning on my devices until I moved the call onto the UI thread. It now initialises correctly
4. your plugin was not intercepting BOOT_COMPLETE in order to initialise Parse. I have added that intent filter to your ParseWakeUpReceiver

I don't know if you are still planning to keep this repo up to date, but thought that these updates might be helpful. Of the 2 Parse Cordova plugin architectures available, I find yours much easier to work with, which is why I have chosen to continue forward development on this one.

Kind regards,
Richard Le Mesurier
Glowworm Software
Cape Town, SA

### CODE REVIEW

**Comment by arnesikstrom** [comment]
Hi,
This repo is still active. It's supporting another plugin for the Construct2 game engine that must work with any changes to this repo. It must support Cordova 4 as that is the main build environment for the users.
Did you test your changes with Cordova 5 or 4?
/Arne

**Comment by mobiRic** [comment]
I am glad it is still active. It was the best starting point for me as a Parse plugin.

I am testing with Ionic 1.6.4, backed by Cordova 5.2.0.

###### Re: Google Play dependencies:

Some time back in the development I was getting an error about "com.google.playservices" no longer being supported, and the framework was automatically trying to fetch "cordova-plugin-googleplayservices" instead.

This is likely due to them moving Cordova plugins into the Node Package Manager repo. However today the old google plugin installs on my system perfectly fine. I must have changed something on my build system.

This is probably not important for your purposes.

###### Re: Cordova 4 compatibility

When compiling with Cordova 5.2.0, I get the following error from Gradle:

```
:compileDebugJava
......\platforms\android\src\se\frostyelk\cordova\parse\plugin\ParsePlugin.java:99: error: cannot find symbol
            webView.post(new Runnable() {
                   ^
  symbol:   method post(<anonymous Runnable>)
  location: variable webView of type CordovaWebView
```

This is due to a change in CordovaWebView, which I think I have tracked down to the 4.0.0 release. They updated the interface so that the webView no longer has a .post() method available.

This is probably something to test carefully on your side. Unfortunately I can't think of a way to support both pre-4 and post-4 situations, without using reflection in the Android code. For my purposes, I'm happy to rule out backwards compatibility, which simplifies things on my side.

###### Re: Parse initialisation on the main thread

Simply put - my devices were hanging with the init process never returning when run from the threadpool.

Since initialisation is a startup cost, I was happy to place it on the main thread as it does not impact my current app.

###### Re: initialisation on BOOT_COMPLETED

The plugin was crashing on reboot since Google Cloud Messaging was trying to initialise, hence Parse, but Parse had not yet been initialised.

Perhaps some devices throw USER_PRESENT before BOOT_COMPLETED, but in my case this was an issue.

I hope this helps.

Let if you want me to break this pull into smaller chunks, although its probably just easier to copy/paste anything helpful on your side, since the changes are pretty minor in each case.

Richard
